app/serializers/event_serializer.rb

```diff
@@ -16,4 +17,25 @@ def address
 
     "#{object.address1} #{object.address2}"
   end
+
+  def distance
+    zip_query = @instance_options[:zip_query]
+    return '' if zip_query.blank?
+
+    get_coordinates(zip_query)
+  end
+
+  def get_coordinates(zip_query)
+    zip_lat = ::ZipCode.select(:latitude)
+                       .where(zip_code: zip_query).to_a.first.latitude
+    zip_long = ::ZipCode.select(:longitude)
+                        .where(zip_code: zip_query).to_a.first.longitude
+    calc_distance(zip_lat, zip_long)
+  end
+
+  def calc_distance(zip_lat, zip_long)
+    Geocoder::Calculations.distance_between([zip_lat, zip_long],
+                                            [object.pt_latitude,
+                                             object.pt_longitude]).round(2)
+  end
```

Comment:
These methods might make more sense in the ZipCode model. They seem pretty zipcode centric and putting them in the model would let you reuse this calculation for distance from and event, foodbank, etc.

Reply:
You've got me thinking.  I'm not sure a generalized solution is ZipCode centric.  We are already thinking about a use case where the UI passes in lat/long from the user's location to the API using lat/long query parameters.  Instead of extracting the zip_code here the controller could have the job of sending the Zipcode lat/long parameters as instance_options, not the zip code.  That would remove the need for a get_coordinate method.  We could do the same for a foodbank, etc.

Reply:
I ended up moving the logic to the event model.  The shared geographic functions are now abstracted in a Geo lib module.